scripts/lib/clean_yarn_cache.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import sys
+
+ZULIP_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.append(ZULIP_PATH)
+from datetime import date, datetime
+
+from dateutil.relativedelta import relativedelta
+
+from scripts.lib.zulip_tools import may_be_perform_purging, parse_cache_script_args, run_as_root
+
+YARN_CACHE_PATH = "../.cache/yarn/"
+CURRENT_VERSION = "v6"
+
+
+def remove_unused_versions_dir(args: argparse.Namespace) -> None:
+    current_version_dir = os.path.join(YARN_CACHE_PATH, CURRENT_VERSION)
+    dirs_to_purge = set(
+        [
+            os.path.join(YARN_CACHE_PATH, directory)
+            for directory in os.listdir(YARN_CACHE_PATH)
+            if directory != CURRENT_VERSION
+        ]
+    )
+
+    may_be_perform_purging(
+        dirs_to_purge,
+        {current_version_dir},
+        "yarn cache",
+        args.dry_run,
+        args.verbose,
+        args.no_headings,
+    )
+
+
+def remove_old_files(dry_run: bool, verbose: bool) -> None:
+    if dry_run:
+        print("Performing a dry run...")
+
+    six_months_ago = date.today() + relativedelta(months=-6)
```

Comment:
You can use `datetime.now()` rather than `date.today()` and that's probably a better choice (in case one wanted to patch this script with a smaller time interval).

Reply:
Also, use a variable name like `date_cutoff` -- there's no reason to hardcode the time interval we picked (which is arbitrary) in the variable name -- that makes it more annoying to change the value.